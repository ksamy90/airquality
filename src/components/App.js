import React from 'react';
import Weather from './SearchWeather';
import apijson from '../data.json';

class App extends React.Component {
    state = {
        apiData: '',
        aqi: '',
        cigg: ''
    }
    onSearchSubmit = (data1) => {
        apijson.map((data2) => {
            if (data1 === data2.name) {
                this.setState(() => ({
                    apiData: data2.name,
                    aqi: data2.aqi,
                    cigg: data2.cigg
                }));
            }
        });
    }
    render() {
        return (
            <div className="main ui container">
                <Weather 
                    onsubmit={this.onSearchSubmit}
                />
                <h2>City: {this.state.apiData}</h2>
                <h3>Air Smog: {this.state.aqi}</h3>
                <h3>Cigarette pollution: {this.state.cigg}</h3>
            </div>
        );
    }
};

export default App;
