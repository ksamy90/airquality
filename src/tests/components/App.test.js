import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App';

test('should render App correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
});

test('should submit correct data', () => {
    const data = {
        name: "Faridabad",
        aqi: "220 PM2.5",
        cigg: "10"
    };
    const wrapper = shallow(<App />);
    wrapper.find('Weather').prop('onsubmit')(data.name);
    expect(wrapper.state('apiData')).toBe(data.name);
    expect(wrapper.state('aqi')).toBe(data.aqi);
    expect(wrapper.state('cigg')).toBe(data.cigg);
});
