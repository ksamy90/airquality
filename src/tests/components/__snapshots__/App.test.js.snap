// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<div
  className="main ui container"
>
  <Weather
    onsubmit={[Function]}
  />
  <h2>
    City: 
  </h2>
  <h3>
    Air Smog: 
  </h3>
  <h3>
    Cigarette pollution: 
  </h3>
</div>
`;
